bcda/cclf/cclf.go

```diff
@@ -551,6 +569,33 @@ func validate(fileMetadata *cclfFileMetadata, cclfFileValidator map[string]cclfF
 	return nil
 }
 
+func validateFileName(fileName string) error {
+	filenameRegexp := regexp.MustCompile(`(T|P).*\.((?:A|T)\d{4})\.ACO.*\.ZC(0|8)Y(\d{2})\.(D\d{6}\.T\d{6})\d`)
+	valid := filenameRegexp.MatchString(fileName)
+	if !valid {
+		fmt.Printf("Invalid filename for file: %s.\n", fileName)
+		err := fmt.Errorf("invalid filename for file: %s", fileName)
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func parseTimestamp(fileMetadata *cclfFileMetadata,fileName string) error {
+	filenameRegexp := regexp.MustCompile(`(T|P).*\.((?:A|T)\d{4})\.ACO.*\.ZC(0|8|9)Y(\d{2})\.(D\d{6}\.T\d{6})\d`)
```

Comment:
With the timestamp being the only part of the filename that we care about in this function, most of the capturing groups could be removed or made non-capturing. Maybe something like this?
`(?:T|P).*\.(?:A|T)\d{4}\.ACO.*\.ZC(?:0|8|9)Y\d{2}\.(D\d{6}\.T\d{6})\d` - date in group 1